### ISSUE TITLE
PayPal REST API

### ISSUE THREAD

**Issue by chriscct7** [awesomemotive/easy-digital-downloads#1958]
We should support the PayPal REST API as soon as we can, as opposed to the current PayPal Standard gateway. The benefits: https://developer.paypal.com/webapps/developer/docs/api/

We'd gain a way to securely store credit cards so users could just come back and checkout

**Comment by chriscct7** [comment]
I thought we updated this. Basically the idea that the core devs have discussed is use the new REST API if 5.3 or up is installed using an if else in the gateway code.

**Comment by chriscct7** [comment]
If we're going forward with #2721, we should really look at a gateway class with gateway supports so that we can easily turn off things gateways don't support, like refunds if 5.2 is on the server

**Comment by chriscct7** [comment]
Maybe do a: "refunds are not available for this gateway" on the view/edit order page or something

**Comment by pippinsplugins** [comment]
Agreed, though let's keep refund discussion on #2721.

**Comment by pippinsplugins** [comment]
This won't happen in 2.4. We'll get Amazon in  but not this one too.

**Comment by pippinsplugins** [comment]
If we do decide to hit this sooner than later, we can do it with the WP_HTTP API. No need to use their SDK.

I worked with it recently for AffiliateWP.

**Comment by cklosowski** [comment]
@pippinsplugins @chriscct7 what's the status on this. Do we need to keep moving forward here? Lots has changed since over a year ago.

**Comment by pippinsplugins** [comment]
We can still add support for this in order to replace PayPal Standard: https://developer.paypal.com/docs/integration/web/accept-paypal-payment/

**Comment by chriscct7** [comment]
The above link is now https://developer.paypal.com/docs/integration/direct/payments/paypal-payments/#

Note, not supporting the REST API causes issues with third party integrations. For example, services like ChartMogul can't accept the PayPal webhook from non REST API implementations

**Comment by clifgriffin** [comment]
@chriscct7 We recently ran into this.

The TLDR is that PayPal is the absolute worst.

We switched to PayPal Express because the recurring profiles it creates are accessible through the PayPal REST API, which allows them to be imported into ChartMogul.

However now we have to figure out how to get the data for 100s of profiles created through PayPal Standard into their system. No easy task.

**Comment by stephywells** [comment]
@clifgriffin Did you find a way to get the PayPal Standard transactions into ChartMogul?

**Comment by clifgriffin** [comment]
@stephywells I ran into a constellation of problems and ultimately shelved the project. 

1. EDD pro-ration is non-standard and ChartMogul doesn't get it
2. EDD upgrades are sequenced such that ChartMogul sees a cancellation and then a new purchase instead of a plan upgrade. 

I spent a bit working on a Baremetrics importer, but ultimately just can't justify the time on this project right now. Using EDD Metrics by Scott Bolinger for now.
